### DISCUSSION TITLE
NextStations issues

### DISCUSSION THREAD

**Discussion by Ironborn65** [erichelgeson/BlueSCSI-v1#125] [Q&A] [answered]
Hi everyone, hi Eric,

I'm supporting Luciano, a friend of mine who has these issues, the NextStations are not mine.
I personally have about 25 Macs but I can not help him, so I'm asking the kind support of the community here.

Machine tested: NeXTstation 68040/25 and NeXTstation Color 68040/25, both with 24 Mb RAM
Schede BLUEscsi: one and the Powerbook version (I can not tell the firmware version), pictures are attached.

![image](https://user-images.githubusercontent.com/24572971/174152689-f210c095-e0f0-4005-b502-aa1a17bb9a4e.png)

OS: NeXTstep 3.1, NeXTstep 3.3, OpenStep 4.0. 

Description of the two issues: 
1) at startup Verbose Mode it identifies 6 LUN SCSI (all internal IDs are identified).
Once the OS is loaded it asks to fox or format these 6 ghosts disks
I can press ignore
In “root” mode the Build Disk app identifies all these bogus disks with “0” bytes, but they are obviously non usable
As a consequence any SCSI device I attach, either internal or external, is not recognised or mounted
 
2) If I switch user  (logout and login) I have a system crash, it shows the spinning wheel forever, I need to force the reboot, this happens with any NextStation and any bluescsi card I have

can you help me please with this?

thanks so much

Pierfranco

I hereby attach two pictures of the verbose mode trace log 

![image](https://user-images.githubusercontent.com/24572971/174153878-481b79c7-0540-4616-b69c-f855bea735a2.png)

![image](https://user-images.githubusercontent.com/24572971/174153907-7610108b-7ba7-4607-b1c6-77e346114af0.png)

**Comment by erichelgeson** [comment, marked as answer]
That looks like you're experiencing a bug in the current firmware with showing multiple LUN's - this will be fixed in the next release - if you could try when we release that (soon) and let us know if it fixes it.

**Comment by marcelv-3** [comment]
don't know what the fix is, but added this to onInquiryCommand to prevent scanning all possible LUNs 

```
byte onInquiryCommand(byte len)
{
  if (m_lun > 0) {  // or > NUM_SCSILUN
    //peripheral qualifier: 011b device on lun not supported
    //peripheral device type: 0x1f unknown or no device type
    SCSI_INFO_BUF[0] = 0x7f;
    LOGN("lun > 0 not supported");
  }
  writeDataPhase(len < 36 ? len : 36, SCSI_INFO_BUF);
  if (m_lun > 0) {
    //reset default info buf answer
    //device type direct access
    SCSI_INFO_BUF[0] = 0x00;
  }
  return SCSI_STATUS_GOOD;
}

```
maybe an usable addtion ?
